src/client.js

```diff
@@ -252,17 +245,8 @@ assign(Client.prototype, {
 
   // Destroy the current connection pool for the client.
   destroy(callback) {
-    const promise = new Promise((resolver) => {
-      if (!this.pool) {
-        return resolver()
-      }
-      this.pool.drain(() => {
-        this.pool.destroyAllNow(() => {
-          this.pool = undefined
-          resolver()
-        })
-      })
-    })
+    const promise = this.pool ? this.pool.destroy() : Promise.resolve()
```

Comment:
Yeah, it was unclear to me what drain did. Currently Tarn murders all connections (both used and free) and calls the destroy function for them. 

I'll change destroy so that it waits for all connections to be released before destroying them.